crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -80,6 +80,7 @@ pub(crate) struct NormalizeTypeOpts {
     pub expand_enum_variant: bool,
 
     pub preserve_keyof: bool,
+    pub in_type_or_type_param: bool,
```

Comment:
Added this because I want to know if I'm inside a type or a type_param. For example, in an assignment `Lowercase<string>` isn't assignable to `Uppercase<string>`

```ts
declare var x: Uppercase<string>
declare var y: Lowercase<string>
x = y ❌
```

but it an type param it is valid
```ts
declare function a(x: Uppercase<Lowercase<string>>) {} ✅
```

Reply:
Makes sense. Thank you!